pkgs/servers/search/typesense/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, nixosTests
+}:
+let
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  sources = lib.importJSON ./sources.json;
+  platform = sources.platforms.${system} or throwSystem;
+  inherit (sources) version;
+  inherit (platform) arch hash;
+in
+stdenv.mkDerivation {
+  pname = "typesense";
+  inherit version;
+  src = fetchurl {
+    url = "https://dl.typesense.org/releases/${version}/typesense-server-${version}-${arch}.tar.gz";
+    inherit hash;
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+  ];
+
+  sourceRoot = ".";
```

Comment:
Why is it needed?

Reply:
Because unfortunately the upstream tar.gz contains no subdirectories, and instead has the files directly on the toplevel

Reply:
I see…

Reply:
Consider adding a comment then! :)

Reply:
Already did that when explaining by source-build attempt. So should exist now :D